src/Compilers/CSharp/Portable/Binder/Binder_NameConflicts.cs

```diff
@@ -51,7 +52,7 @@ private static Location GetLocation(Symbol symbol)
                     {
                         // Type parameter declaration name conflicts are detected elsewhere
                     }
-                    else
+                    else if (!isLocalFunction)
```

Comment:
I found this surprising. Should we just change the behavior for static local functions?

Reply:
I believe the expected behavior is to allow shadowing for all local functions.

---
In reply to: [244605104](https://github.com/dotnet/roslyn/pull/32067#discussion_r244605104) [](ancestors = 244605104)

Reply:
Part of the motivation for the shadowing decisions was that I should be able to take a normal function and copy + paste it as a local function and it should just compile. No hassle. Given that I would expect all forms of shadowing to work. 

This is my memory of the meeting at least.